src/api/l_audio.c

```diff
@@ -65,21 +67,88 @@ static int l_lovrAudioNewSource(lua_State* L) {
   return 1;
 }
 
+static int l_lovrAudioSetListenerPose(lua_State *L) {
+  float position[4], orientation[4];
+  int index = 1;
+  index = luax_readvec3(L, index, position, NULL);
+  index = luax_readquat(L, index, orientation, NULL);
+  lovrAudioSetListenerPose(position, orientation);
+  return 0;
+}
+
+static int l_lovrAudioGetCaptureDuration(lua_State *L) {
+  TimeUnit units = luax_checkenum(L, 1, TimeUnit, "seconds");
+  size_t sampleCount = lovrAudioGetCaptureSampleCount();
+  
+  if (units == UNIT_SECONDS) {
+    lua_pushnumber(L, (double) sampleCount / LOVR_AUDIO_SAMPLE_RATE);
+  } else {
+    lua_pushinteger(L, sampleCount);
+  }
+  return 1;
+}
+
+static int l_lovrAudioCapture(lua_State* L) {
+  int index = 1;
+  
+  size_t samples = lua_type(L, index) == LUA_TNUMBER ? lua_tointeger(L, index++) : lovrAudioGetCaptureSampleCount();
+
+  if (samples == 0) {
+    return 0;
+  }
+
+  SoundData* soundData = luax_totype(L, index++, SoundData);
+  size_t offset = soundData ? luaL_optinteger(L, index, 0) : 0;
+
+  if (soundData) {
+    lovrRetain(soundData);
+  }
+
+  soundData = lovrAudioCapture(samples, soundData, offset);
+  luax_pushtype(L, SoundData, soundData);
+  lovrRelease(SoundData, soundData);
+  return 1;
+}
+
+static int l_lovrAudioSetCaptureDevice(lua_State *L) {
+  //
+
+  return 0;
+}
```

Comment:
I was planning on not exposing device iteration and always using the default device, unless people started asking for it. 
 For VR games and utilities it seems most common to use the single microphone built into the headset.

If this is added though, maybe it could be an optional table passed to `start` indicating desired device and settings for stuff like format/sample rate/buffer size preferences?

While I was still planning on doing device iteration I had a function like this:

```lua
for id, name, type, minChannels, maxChannels in lovr.audio.devices() do

end
```

Reply:
Alright. I really really need device iteration for Alloverse, so I'll need it in our fork anyway. I'll submit a suggestion and you can take it out in mainline if you want. (though in my experience, automatic device selection can work really poorly in PCVR titles outside of Oculus headsets, and taking off headset to switch recording device can be a hassle).

that's an interesting API! I can adapt something similar :)

Reply:
Ok, it's helpful to know that you need multiple devices, that makes it a higher priority.

Reply:
I don't need to connect to multiple devices at once, but I need to be able to switch between them (so I can present a mic selector in my app), just to be clear :)

Reply:
@bjornbytes would `for _, deviceDescription in ipairs(lovr.audio.devices()) ` with `deviceDescription` being a table with name, type, etc be ok, worse, better?

Reply:
That's probably better.  If it returns a table it ought to be called `getDevices` though.